src/app/shared/formatters/app-currency.pipe.ts

```diff
@@ -7,24 +7,26 @@ import { LocaleService } from '../../services/locale.service';
 @Pipe({ name: 'appCurrency' })
 export class AppCurrencyPipe implements PipeTransform {
   private currencyPipe!: CurrencyPipe;
-  private currency!: string;
-
+  private currencyCode!: string; // The [ISO 4217] currency code
   public constructor(
     private centralServerService: CentralServerService,
     private localeService: LocaleService) {
     // Get the locale
     this.localeService.getCurrentLocaleSubject().subscribe((locale) => {
       this.currencyPipe = new CurrencyPipe(locale.currentLocaleJS);
     });
-    // Get the Pricing settings
-    this.currency = this.centralServerService.getLoggedUser().currency;
+    // Get the currency code
+    // The currency code is part of the user token and comes from the Pricing Settings!
+    this.centralServerService.getCurrentUserSubject().subscribe((user) => {
+      // user is null when logging out
+      this.currencyCode = user?.currency;
```

Comment:
I don't see why we need to subscribe here?

As discussed, why not using an Angular service to pull that value from the backend.
I don't know if we have similar use case with other variables.

And as discussed with Caroline, I didn't want to handle the logout from the backend.

Maybe the simplest solution would be that when the user changes the currency + click on Save -> Display a warning message that tells that it will be logged out + Customer accepts + Call backend to save the currency + Force logout.

Reply:
> With the previous code, the logout had no impact. The pipe instance was not recreated after a new logon and the previous currency code was still used. This is the reason why have changed the logic.

Have you checked how Angular handles pipe and how dependency injection can influence the lifetime of it?

> Now, each **AppCurrencyPipe** gets the up-to-date currency code (from the user token) at log on. This is fast enough because the token is parsed only once and kept in the **centralServerService**.

Okay, but to get the new token, you'll need to logout the user, right?

Reply:
> I have a question: Why would you subscribe for the localeJs and not for the currency code? Both info comes from the token. So we should have the same logic for both. Do you agree?

It's not because I made a choice 2 years ago that it's still the right one today after 4 major Angular releases ;-).
Why do we cache the currency in the pipe, why don't we always use the centralServerService to get the currency?
What would be the overhead?

Reply:
Remove the class members and always call the service?
Did you test this?
<img width="732" alt="2022-01-19_12-17-35" src="https://user-images.githubusercontent.com/25266945/150119865-1054737f-5ffa-41da-9fbf-a0018bba8935.png">
This would remove also the locale subscription.